tests/integration/helpers.py

```diff
@@ -175,6 +177,26 @@ async def get_route_condition(ops_test: OpsTest, route_name: str) -> Optional[Di
         return None
 
 
+async def get_hpa(ops_test, hpa_name: str) -> Optional[HorizontalPodAutoscaler]:
+    """Retrieve the HPA resource so we can inspect .spec and .status directly.
+
+    Args:
+        ops_test: pytest-operator plugin
+        hpa_name: Name of the HPA resource.
+
+    Returns:
+        The HorizontalPodAutoscaler object or None if not found / on error.
+    """
+    model = ops_test.model.name
+    try:
+        c = lightkube.Client()
+
+        return c.get(HorizontalPodAutoscaler, namespace=model, name=hpa_name)
+    except Exception as e:
+        logger.error("Error retrieving HPA %s: %s", hpa_name, e, exc_info=True)
+        return None
```

Comment:
nit: `ops_test` is only used to get `ops_test.model.name`, so just name the arg namespace

```suggestion
async def get_hpa(namespace: str, hpa_name: str) -> Optional[HorizontalPodAutoscaler]:
    """Retrieve the HPA resource so we can inspect .spec and .status directly.

    Args:
        namespace: Namespace of the HPA resource.
        hpa_name: Name of the HPA resource.

    Returns:
        The HorizontalPodAutoscaler object or None if not found / on error.
    """
    try:
        c = lightkube.Client()

        return c.get(HorizontalPodAutoscaler, namespace=namespace, name=hpa_name)
    except Exception as e:
        logger.error("Error retrieving HPA %s: %s", hpa_name, e, exc_info=True)
        return None
```